File: ExamWebApp/ExamDoorWay/src/main/java/com/exam/service/AttemptService.java
package com.exam.service;

import java.util.List;
import java.util.Set;

import com.exam.model.User;
import com.exam.model.exam.Attempt;
import com.exam.model.exam.Category;
import com.exam.model.exam.Quiz;

public interface AttemptService {

	public Attempt addAttempt(Attempt attempt);

	public Attempt getAttempt(Long attemptId);
	
	public Attempt updateAttempt(Attempt attempt);
	
	public Set<Attempt> getAttempts();

	public void deleteAttempt(Long attemptId);

	public List<Attempt> getAttemptByUserId(User user);
	
	public List<Attempt> getAttemptByQuizId(Quiz quiz);

}

File: ExamWebApp/ExamFrontend/src/app/services/attempt.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import baseUrl from './helper';

@Injectable({
  providedIn: 'root'
})
export class AttemptService {

  constructor(private _http:HttpClient) { }

  //adding a question
  public addAttempt(attempt:any) {
    return this._http.post(`${baseUrl}/attempt/addAttempt`, attempt)
  }

  //getting attempt by user id
  public getAttemptByUserId(userId:any) {
    return this._http.get(`${baseUrl}/attempt/getAttemptByUserId/${userId}`)
  }

  //getting attempt by quiz id
  public getAttemptByQuizId(quizId:any) {
    return this._http.get(`${baseUrl}/attempt/getAttemptByQuizId/${quizId}`)
  }
  
  // deleting attempt
  public deleteAttempt(attemptId:any) {
    return this._http.delete(`${baseUrl}/attempt/deleteAttempt/${attemptId}`)
  }

}
